tests/functional/__init__.py

```diff
@@ -10,3 +10,31 @@
 # distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
 # ANY KIND, either express or implied. See the License for the specific
 # language governing permissions and limitations under the License.
+import os
+import uuid
+
+_ORIGINAL = os.environ.copy()
+# These are environment variables that allow users to control
+# the location of config files used by botocore.
+_CONFIG_FILE_ENV_VARS = [
+    'AWS_CONFIG_FILE',
+    'AWS_SHARED_CREDENTIALS_FILE',
+    'BOTO_CONFIG',
+]
+_CREDENTIAL_ENV_VARS = [
+    'AWS_ACCESS_KEY_ID',
+    'AWS_SECRET_ACCESS_KEY',
+    'AWS_SESSION_TOKEN',
+]
+
+
+def setup_package():
+    random_file = str(uuid.uuid4())
+    for varname in _CONFIG_FILE_ENV_VARS:
+        os.environ[varname] = random_file
```

Comment:
> Isn't this whole thing going to get ignored because BaseSessionTest monkeypatches os.environ in its setUp method?

For the specific test I fixed, yes.

The intent of these test fixtures is to address how this can caught in the future.  That is, how can people know they need to use `BaseSessionTest` or face potential travis failures?  If you have credentials configured in your dev setup (very likely) you'll never see this issue until travis runs the tests.  This forces the issue locally by setting all the normal cred env vars to bogus values so you'll see similar behavior to what travis sees.